caf-correlation-id/pom.xml

```diff
@@ -0,0 +1,161 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+
+    Copyright 2019-2020 Micro Focus or one of its affiliates.
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+         http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+
+-->
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <parent>
+        <artifactId>caf-logging</artifactId>
+        <groupId>com.github.cafapi.logging</groupId>
+        <version>1.4.0-SNAPSHOT</version>
+    </parent>
+    <modelVersion>4.0.0</modelVersion>
+    
+    <artifactId>caf-correlation-id</artifactId>
+    
+    <dependencies>
+        <dependency>
+            <groupId>io.dropwizard</groupId>
+            <artifactId>dropwizard-core</artifactId>
+            <scope>provided</scope>
```

Comment:
I split the project into separate modules in 9a26ccd to resolve this.

Reply:
I have used `scope=provided` for below mentioned reasons -
1. `CorrelationIdFilter` and `CorrelationIdClientFilter`- These filters implement interfaces `javax.servlet.Filter` and `javax.ws.rs.client.ClientRequestFilter` respectively. The implementation jar varies from project to project. Thus, found it better in this case for consumer to own the implementation jar dependency rather than the library. 
2. To avoid a version mismatch on dependencies. The code in this library does not require any specific version of dropwizard or spring and thus leaving it to the consumer to provide the dependent jars via `provided`.

These dependencies are not optional and thus did not use optional scope. I did start without `provided` scope in the beginning but ran into the above mentioned issues. These issues were reported by analyze plugins and were causing the build to fail. 
I see you kept the provided scope for the `CorrelationIdFilter` but did not do so for the `CorrelationIdClientFilter`. If you are in agreement, let me know I will add the same.

Reply:
As discussed, this comment was relating to `dropwizard-core`.  I didn't remove `scope=provided` from `jakarta.servlet-api`.  I wasn't sure about `jakarta.ws.rs-api` so feel free to re-add it to that dependency if it's causing a problem.